Logstash/logstash-input-azuretopicthreadable/lib/logstash/inputs/azuretopicthreadable.rb

```diff
@@ -47,9 +58,12 @@ def process(output_queue, pid)
           codec.decode(message.body) do |event|
           decorate(event)
           output_queue << event
-        end # codec.decode
-        # delete the message after reading it
-        azure_service_bus.delete_subscription_message(message)
+          # delete the message after reading it
+          azure_service_bus.delete_subscription_message(message)
+        end
+        if !message
+          sleep(@thread_sleep_time) #topic is probably empty. sleep. 
```

Comment:
It is good to put a sleep when the topic is empty.
However, in case the @thread_sleep_time been set long. During the sleeping, it might worth to check the stop signal:
```
Stud.stoppable_sleep(@thread_sleep_time) { stop? }
```